python/ray/air/_internal/tensorflow_utils.py

```diff
@@ -81,3 +81,63 @@ def tensorize(series):
         return tf.expand_dims(concatenated_tensor, axis=1)
 
     return concatenated_tensor
+
+
+def convert_ndarray_to_tf_tensor(
+    ndarray: np.ndarray,
+    dtype: Optional[tf.dtypes.DType] = None,
+) -> tf.Tensor:
+    """Convert a NumPy ndarray to a TensorFlow Tensor.
+
+    Args:
+        ndarray: The NumPy ndarray that we wish to convert to a TensorFlow Tensor.
+        dtype: The TensorFlow dtype for the created tensor; if None, the dtype will be
+            inferred from the NumPy ndarray data.
+
+    Returns: A TensorFlow Tensor.
+    """
+    tf_tensor = tf.convert_to_tensor(ndarray, dtype=dtype)
+
+    # Off-the-shelf Keras Modules expect the input size to have at least 2
+    # dimensions (batch_size, feature_size). If the tensor for the column
+    # is flattened, then we unqueeze it to add an extra dimension.
+    if len(tf_tensor.shape) == 1:
+        tf_tensor = tf.expand_dims(tf_tensor, axis=1)
```

Comment:
I'm still uneasy about doing this magic conversion under the hood. Could we at least make it an optional flag, such as `auto_unsqueeze_flat_tensors = True`?

Reply:
This ends up becoming the current `to_torch` and `to_tf`...

Reply:
I see. I have the following proposal then. How about we simplify iter_torch_batches even further and completely remove these magic conversions, as well as the dict vs non-dict scenario.

In short:
- iter_X_batches will behave structurally identical to iter_batches()
- suppose you have a single column "foo", the user will have to explicitly `model(batch["foo"].unsqueeze())`. I think this is fine and well worth the tradeoff for simplification.

Reply:
Ok then lets just remove the unsqueezing altogether, and have users unsqueeze as part of the preprocessor logic.

I really do think we should have the exact same behavior for training and prediction, otherwise it can be confusing/frustrating to users (why does my model work with training, but not for prediction?)

Reply:
Makes sense. The user could unsqueeze in a BatchMapper, that's actually a good example preprocessor to have in basic example guides.

Reply:
ok sounds good- sorry for the back and forth on this @clarkzinzow!

Reply:
No worries! I'm much happier with this resolution. 😄